src/hotspot/share/runtime/os.hpp

```diff
@@ -310,13 +311,21 @@ class os: AllStatic {
                                                   const size_t size);
 
   static int    vm_allocation_granularity();
-  static char*  reserve_memory(size_t bytes, char* addr = 0,
-                               size_t alignment_hint = 0, int file_desc = -1);
-  static char*  reserve_memory(size_t bytes, char* addr,
-                               size_t alignment_hint, MEMFLAGS flags);
+
+  // Reserves virtual memory.
+  // alignment_hint - currently only used by AIX
+  static char*  reserve_memory(size_t bytes, size_t alignment_hint = 0, MEMFLAGS flags = mtOther);
+
+  // Reserves virtual memory.
+  // if file_desc != -1, also attaches the memory to the file.
+  static char*  reserve_memory_with_fd(size_t bytes, size_t alignment_hint, int file_desc);
+
```

Comment:
I like that you split those two apart. The fd parameter in reserve_memory really bugged me. Should we do this also for attempt_reserve_memory_at?

Reply:
Yes, I was thinking about that, but wanted to slice this into easily reviewed parts.

Reply:
Created JDK-8253652.